spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -322,6 +322,12 @@ public void onSuccess(SendResult<byte[], byte[]> result) {
 		props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
 		props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
 		props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, 100);
+		
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, anonymous ? "latest" : "earliest");
+		if (!ObjectUtils.isEmpty(consumerProperties.getExtension().getStartOffset())) {
```

Comment:
Sure, I can do that.  I wanted the anonymous check logic to be before though because I would think that would be the least likely way you'd want the property set (basically set it that way in absence of any other mechanism).

I'll move the binder config check to the end of the method and see what you think.  Thanks for the feedback!